ui/src/OpenSearch.tsx

```diff
@@ -23,105 +23,145 @@ const getOpenSearchVersion = async () => {
+            //keyword
+            query.query.hybrid.queries[0].bool.filter = filters["keyword"]
+        }
+        if (query.query.hybrid.queries && query.query.hybrid.queries.length > 0 && query.query.hybrid.queries[1].neural) {
+            //neural
+            query.query.hybrid.queries[1].neural.embedding.filter = filters["neural"]
+        }
+    }
+    return query
+}
+
+export const hybridConversationSearchNoRag = async (rephrasedQuestion: string, filters: any, index_name: string, model_id: string) => {
+    const { osQuery, url } = getHybridSearchNoRagQuery(rephrasedQuestion, filters, index_name, model_id)
+    return openSearchCall(osQuery, url)
+}
+
+export const getHybridSearchNoRagQuery = (rephrasedQuestion: string, filters: any, index_name: string, model_id: string) => {
+    let osQuery = hybridSearch(rephrasedQuestion, filters, index_name, model_id)
     const url = "/opensearch/" + index_name + "/_search?search_pipeline=" + NO_RAG_SEARCH_PIPELINE
-    return openSearchCall(query, url)
+    return { osQuery, url }
 }
 
-export const hybridConversationSearch = async (question: string, rephrasedQuestion: string, conversationId: string, index_name: string, embeddingModel: string, llmModel: string, numDocs: number = 7) => {
-    let genQaParams: {[key: string]: any} = {
-        "llm_question": question,
-        "context_size": numDocs,
-        "llm_model": llmModel,
-    }
-    if(await is2dot12plus()) {
-        genQaParams.memory_id = conversationId;
-    } else {
-        genQaParams.conversation_id = conversationId;
+export const getHybridSearchQuery = (question: string, rephrasedQuestion: string, filters: any, index_name: string, embeddingModel: string, llmModel: string, numDocs: number = 7, rerank: boolean = false) => {
+    let query: any = hybridSearch(rephrasedQuestion, filters, index_name, embeddingModel)
+    query.ext = {
+        "generative_qa_parameters": {
+            "llm_question": question,
+            "context_size": numDocs,
+            "llm_model": llmModel,
+        }
     }
-    const query =
-    {
-        "_source": SOURCES,
-        "query": {
-            "hybrid": {
-                "queries": [
-                    {
-                        "match": {
-                            "text_representation": rephrasedQuestion
-                        }
-                    },
-                    {
-                        "neural": {
-                            "embedding": {
-                                "query_text": rephrasedQuestion,
-                                "k": 100,
-                                "model_id": embeddingModel
-                            }
-                        }
-                    }
-                ]
+    if (rerank) {
+        query.ext.generative_qa_parameters["rerank"] = {
+            "query_context": {
+                "query_text": rephrasedQuestion
```

Comment:
oh. surely there's a way to make all this a little more concise?

Reply:
Cleaned it up a bit, removed the non-conversation query execution since we aren't using it. We still need the 3 ops: getconverstationquery, getnonconversationquery, executeconversation